flow-typed/src/ChorusClient.js

```diff
@@ -0,0 +1,12 @@
+interface ChorusClient {
+	open(url: string): Promise<Event>,
+	close(): void,
+	connect(): void,
+	publishStep(
+		pattern: string,
+		callback: (...args: Array<string>) => number | string | void,
```

Comment:
Regarding the automatic type coersion, I did read that in your docs and caught a glimpse in your Java source code too. Not sure how much of it will be possible with JavaScript yet, as you correctly mentioned, at runtime we won't have any type information anymore (aside from basic primitive types). I'm also not really familiar with what *reflection* is. For the context yeah, no problem, we can iterate.